libs/design/switch/src/switch/switch.component.ts

```diff
@@ -7,14 +9,80 @@ import {
   Output,
 } from '@angular/core';
 
+import { DAFF_LOADING_ICON_COMPONENTS } from '@daffodil/design/loading-icon';
+import { DaffToastService } from '@daffodil/design/toast';
+
 @Component({
   selector: 'daff-switch',
   changeDetection: ChangeDetectionStrategy.OnPush,
   templateUrl: './switch.component.html',
   styleUrls: ['./switch.component.scss'],
   standalone: true,
+  imports: [
+    CommonModule,
+    DAFF_LOADING_ICON_COMPONENTS,
+  ],
 })
 export class DaffSwitchComponent {
   @Input() checked = false;
+  @Input() loading = false;
+  @Input() position: 'left' | 'right' | 'top' | 'bottom' = 'left';
+
   @Output() toggle = new EventEmitter<boolean>();
+
+  constructor(private toastService: DaffToastService) {}
+
+  onToggle() {
+    try {
+      this.checked = !this.checked;
+      this.toggle.emit(this.checked);
+
+    } catch (error) {
+      this.toastService.open({
+        title: 'Toggle Error',
+        message: error.message,
+        status: 'danger',
+      },
+      {
+        duration: 5000,
+      });
+    }
+  }
+
+  @HostBinding('class.daff-switch--position-left') get isLeftPosition() {
```

Comment:
These `HostBindings` are unnecessary if the class is being applied to the `<label class="daff-switch">` element.